File: backend-api/reservation-api/src/main/java/com/hcmute/reservation/model/dto/table/AvailableWindowResponse.java
package com.hcmute.reservation.model.dto.table;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class AvailableWindowResponse {

    public enum Availability { FULL_AVAILABLE, PARTIAL_AVAILABLE }

    private Long tableId;
    private int capacity;
    private Availability availability;
    // non-null nếu PARTIAL_AVAILABLE — thời điểm bàn sẽ trống
    private LocalDateTime availableUntil;
    // Các bàn có thể ghép để đủ chỗ
    private List<Long> mergeCandidateIds;
}

File: desktop-staff/reservation-winforms/DTO/table/AvailableWindowResponse.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace reservation_winforms.DTO.table
{
    public class AvailableWindowResponse
    {
        [JsonProperty("tableId")]
        public long? TableId { get; set; }

        [JsonProperty("capacity")]
        public int Capacity { get; set; }

        [JsonProperty("availability")]
        public string Availability { get; set; } // "FULL_AVAILABLE" hoặc "PARTIAL_AVAILABLE"

        [JsonProperty("availableUntil")]
        public DateTime? AvailableUntil { get; set; }

        [JsonProperty("mergeCandidateIds")]
        public List<long> MergeCandidateIds { get; set; }
    }
}
